### PR TITLE
Add I2C, UART0 and UART1 pins for KNoT Mesh

### PR DESCRIPTION
**Pull request by luigiluz**
Change knot_mesh.dts file to support i2c, uart0 and uart1 peripherals for 
KNoT Mesh board.

### CODE REVIEW

File: boards/arm/knot_mesh/knot_mesh.dts [hunk 1]
```diff
@@ -76,12 +76,26 @@
 	status ="ok";
 };
 
+/* Map UART0 to unused pins to avoid conflits with logger in
```
Comment (vitbaq) [inline boards/arm/knot_mesh/knot_mesh.dts]: Jump on line.

```
/**
 * Your text here...
 * ...
 */
```

File: boards/arm/knot_mesh/knot_mesh.dts [hunk 2]
```diff
@@ -90,8 +90,8 @@
 
 &i2c0 {
 	status = "ok";
-	sda-pin = <58>;
-	scl-pin = <59>;
+	sda-pin = <26>;
```
Comment (vitbaq) [inline boards/arm/knot_mesh/knot_mesh.dts:108]: In the commit message 1b9bcdb6ff16236f621d58d323d0781411461ae7

You are not adding default pins in this commit. You are changing. Please, rewrite the commit message to be more clear.

**Review by vitbaq — CHANGES_REQUESTED** [review]

File: boards/arm/knot_mesh/knot_mesh.dts (hunk 1 above)
Comment (luigiluz) [reply boards/arm/knot_mesh/knot_mesh.dts]: Done

File: boards/arm/knot_mesh/knot_mesh.dts (hunk 2 above)
Comment (luigiluz) [reply boards/arm/knot_mesh/knot_mesh.dts:108]: Done

**Review by vitbaq — APPROVED** [review]

**Review by alla3 — APPROVED** [review]

**Pull request merged**